### PR TITLE
Prepare for making persistence layer async.

### PR DESCRIPTION
**Pull request by baldersheim**
Avoid state in the thread.

@vekterli PR

### CODE REVIEW

File: storage/src/vespa/storage/persistence/persistencethread.cpp [hunk 1]
```diff
@@ -934,7 +933,10 @@ PersistenceThread::flushAllReplies(
         }
 #endif
         spi::Bucket b(bucket, spi::PartitionId(_env._partition));
-        spi::Result result = _spi.flush(b, _context);
+        // Flush is not used for anything currentlu, and the context is not correct either when batching is done
+        //So just faking it here.
```
Comment (vekterli) [inline storage/src/vespa/storage/persistence/persistencethread.cpp:937]: Super nitpick: missing leading space

File: storage/src/vespa/storage/persistence/persistencethread.cpp [hunk 2]
```diff
@@ -186,9 +185,9 @@ PersistenceThread::handleGet(api::GetCommand& cmd)
     document::FieldSetRepo repo;
     document::FieldSet::UP fieldSet = repo.parse(*_env._component.getTypeRepo(), cmd.getFieldSet());
     // _context is reset per command, so it's safe to modify it like this.
-    _context.setReadConsistency(api_read_consistency_to_spi(cmd.internal_read_consistency()));
+    context.setReadConsistency(api_read_consistency_to_spi(cmd.internal_read_consistency()));
```
Comment (vekterli) [inline storage/src/vespa/storage/persistence/persistencethread.cpp:188]: Consider removing now outdated comment for this line regarding old `_context`

**Review by vekterli — APPROVED** [review]
:+1: to make sure I recall our discussions on this correctly, the idea is that we will introduce an optional completion callback to the context that can be used by operations supporting async completion?

File: storage/src/vespa/storage/persistence/persistencethread.cpp (hunk 1 above)
Comment (baldersheim) [reply storage/src/vespa/storage/persistence/persistencethread.cpp]: Fixed

File: storage/src/vespa/storage/persistence/persistencethread.cpp (hunk 2 above)
Comment (baldersheim) [reply storage/src/vespa/storage/persistence/persistencethread.cpp:187]: Fixed

**Pull request merged**